doc/running-simple-app.md

```diff
@@ -0,0 +1,196 @@
+
+Creating a Simple Application Image
+----------------------
+
+This section explains how to create single-file images for use with
+Mystikos.
+
+### CPIO
+
+The following command creates an EXT2 image (``cpioimage``) from the
+contents of a directory (``appdir``).
+
+```
+$ myst mkcpio appdir cpioimage
+```
+### EXT2
+
+The following command creates an EXT2 image (``ext2image``) from the
+contents of a directory (``appdir``).
+
+```
+$ myst mkext2 appdir ext2image
+```
+
+> To create a signed EXT2 image to use with a signed application package, see [sign-package.md](./sign-package.md)
+
+Loading a root file system
+--------------------------------
+
+This section explains how to use an application image as a Mystikos root file system
+(or rootfs).
+
+### Using an application image during development
+During development, an application image is passed to the ``myst exec`` command as
+follows.
+
+```
+$ myst exec-sgx <fsimage> /bin/hello
+Hello World!
+```
+
+> `<fsimage>` can be either a CPIO or EXT2 image
+
+
+Advanced Features
+------------
+## Creating an encrypted EXT2 image 
+
+The following command creates an encrypted EXT2 image.
+
+```
+$ myst mkext2 --encrypt=keyfile ext2image
+```
+
+Use the following command generate a test keyfile.
+
+```
+$ head -c 64 /dev/random > keyfile
+```
+
+The EXT2 image created above can be decrypted by a Mystikos image that possesses
+the key. Key distribution is left to the application. An encrypted image cannot
+be used as the root file system.
+
+##  Mounting EXT2 images from applications (explicit mount or "app mounted")
+
+Applications may mount EXT2 file systems with the Linux ``mount()`` function as
+delcared here.
+
+```
+    int mount(
+        const char* source,
+        const char* target,
+        const char* filesystemtype,
+        unsigned long mountflags,
+        const void* data);
+```
+
+The ``source`` is the path of a directory on the trusted root file system where
+the EXT2 file system will be mounted. The ``target`` is the path of the EXT2
+image on the untrusted file system. The ``filesystemtype`` must be "ext2".
+Consider the following example:
+
+```
+    mount(source, "/mnt/mydisk", "ext2", 0, NULL);
+```
+
+This mounts ``source`` onto the "/mnt/mydisk" directory. As before,
+``source`` must refer to an EXT2 image that was generated with by the
+``myst mkext2`` command (it must contain the hash tree and file-system
+signature structure).
+
+The application should be run (``myst exec-sgx``) or packaged
+(``myst packaget-sgx``) with the ``--roothash`` or ``--pubkey`` option as
+explained above (otherwise moutning will fail verification).
+
+To mount an encrypted EXT2 image, the application passes the key to the
+``mount()`` function as shown below.
+
+```
+    const char* args[] =
+    {
+        "key",
+        key,
+        NULL,
+    };
+
+    mount(source, "/mnt/mydisk", "ext2", 0, &args);
+```
+
+Securely obtaining the key form a trusted source is left to the application.
```

Comment:
Point to the SSR sample over here.

Reply:
Added reference to `mystikos/solutions/mhsm_ssr`.